support/documentation/architecture/decisions/0004-project-structure.md

```diff
@@ -0,0 +1,39 @@
+# Project structure
+
+- Status: accepted
+- Date: 2020-05-11
+
+## Context and Problem Statement
+
+We need a consistent way of structuring code in the project. How should we structure our code?
+
+## Decision Outcome
+
+Our project should consist of multiple modules where each of the modules will have only one clearly defined purpose. Our modules will consist of three separate modules (where applicable, since not all
+the modules require such a separation), the `api` module will contain public interfaces, classes for method parameters and results of those interfaces, `core` module containing the implementation that
+depends on the api module and `spring-boot` module that contains Spring Boots autoconfiguration for specified and has dependencies on both `api` and `core` modules. Services in `core` module
+implementing the interface from the `api` module should have a `Default` prefix. `spring-boot` module sole responsibility should only be providing the configuration for `core` module (the
+configuration should consist of an `@Configuration` class and a `@ConfigurationProperties` class (if required) that will provide defaults for property value) while allowing for overriding of defined
```

Comment:
Either 

an `@Configuration` class and an `@ConfigurationProperties`

or 

a `@Configuration` class and a `@ConfigurationProperties`

`a` could be used relative to the `C`, but `an` makes more sense relative to `@`